core/widgetdiv.ts

```diff
@@ -145,6 +161,19 @@ export function hideIfOwner(oldOwner: unknown) {
     hide();
   }
 }
+
+/**
+ * Destroy the widget and hide the div if it is being used by an object in the
+ * specified workspace.
+ *
+ * @param oldOwnerWorkspace The workspace that was using this container.
+ */
+export function hideIfOwnerIsInWorkspace(oldOwnerWorkspace: WorkspaceSvg) {
+  if (ownerWorkspace === oldOwnerWorkspace) {
```

Comment:
I think this should also hide if `ownerWorkspace` doesn't exist. That way for backwards compatibility if someone doesn't set the owner, we default to the old behavior of hiding it anyway.

Reply:
Good idea! Done.